dashboard-api/internal/platform/server/handler/studyconf/read.go

```diff
@@ -0,0 +1,62 @@
+package studyconf
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	studiesmanager "github.com/vlab-research/vlab/dashboard-api/internal"
+	"github.com/vlab-research/vlab/dashboard-api/internal/platform/server/middleware/auth"
+	"github.com/vlab-research/vlab/dashboard-api/internal/platform/storage"
+	"github.com/vlab-research/vlab/dashboard-api/internal/types"
+)
+
+type readResponse struct {
+	Data types.StudyConf `json:"data"`
+}
+
+// ReadHandler is a gin handler used to retrieve a studies configuration
+func ReadHandler(r storage.Repositories) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		var req struct {
+			Slug string `uri:"slug" binding:"required"`
+		}
+		if err := ctx.ShouldBindUri(&req); err != nil {
+			ctx.Status(http.StatusBadRequest)
+			return
+		}
+
+		s, err := r.Study.GetStudyBySlug(ctx, req.Slug, auth.GetUserIdFrom(ctx))
+		if err != nil {
+			handleError(ctx, err)
+			return
+		}
+
+		dscs, err := r.StudyConf.GetByStudyID(ctx, s.Id)
```

Comment:
Why not just create a GetByStudySlug (and user)? Seems more efficient to just put that join in the query, no?

Reply:
True, I can update that